### ISSUE TITLE
Support for Falcon 180B

### ISSUE THREAD

**Issue by Kunjesh07** [oobabooga/text-generation-webui#3867]
**Description**
Have you guys added support for Falcon 180B model ? TheBloke/Falcon-180B-Chat-GPTQ
It is not working in as of now in UI

**Comment by Ph0rk0z** [comment]
What does it say? Any error? It may have to run with trust remote code enabled. AutoGPTQ has support last I heard.

**Comment by Kunjesh07** [comment]
> What does it say? Any error? It may have to run with trust remote code enabled. AutoGPTQ has support last I heard.

t == DeviceType::CUDAINTERNAL ASSERT FAILED at "../c10/cuda/impl/CUDAGuardImpl.h":24, please report a bug to PyTorch.

**Comment by Kunjesh07** [comment]
> I tried to load GGUF Falcon 180B Q8 it is loading just fine so far.

Did you get the output ? like have you asked the question and does it gave you the output ?

**Comment by Kunjesh07** [comment]
> That's weird. Try the GGUF version, are you able to GPTQ the whole thing?

I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model

**Comment by Flanua** [comment]
> > That's weird. Try the GGUF version, are you able to GPTQ the whole thing?
> 
> I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model

My Obaboga was installed without GPU support in CPU mode only though. Sorry I didn't had time to ask model a question yet because it would take to much time to generate the first response plus I'm using swap file for more ram because I have only 128GB. But I will defiantly try and test this model soon I'm very curious to see how it behave.

**Comment by Flanua** [comment]
> > That's weird. Try the GGUF version, are you able to GPTQ the whole thing?
> 
> I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model

If you load this model in CPU you should install Obaboga in CPU mode only and you should use GGUF model variant for CPU mode.

**Comment by Kunjesh07** [comment]
> > > That's weird. Try the GGUF version, are you able to GPTQ the whole thing?
> > 
> > 
> > I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model
> 
> If you load this model in CPU you should install Obaboga in CPU mode only and you should use GGUF model variant for CPU mode.

GGUF is too large. And how to load obaboga in CPU ?

**Comment by Flanua** [comment]
> > > > That's weird. Try the GGUF version, are you able to GPTQ the whole thing?
> > > 
> > > 
> > > I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model
> > 
> > 
> > If you load this model in CPU you should install Obaboga in CPU mode only and you should use GGUF model variant for CPU mode.
> 
> GGUF is too large. And how to load obaboga in CPU ?

Install Obaboga from a scratch and select to install it when isked with CPU mode only.

**Comment by Kunjesh07** [comment]
> > > > > That's weird. Try the GGUF version, are you able to GPTQ the whole thing?
> > > > 
> > > > 
> > > > I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model
> > > 
> > > 
> > > If you load this model in CPU you should install Obaboga in CPU mode only and you should use GGUF model variant for CPU mode.
> > 
> > 
> > GGUF is too large. And how to load obaboga in CPU ?
> 
> Install Obaboga from a scratch and select to install it when isked with CPU mode only.

Are you a rookie in this? We just need to clone the repository and start the server file. There's no option to load it in CPU.

**Comment by Flanua** [comment]
> > > > > > That's weird. Try the GGUF version, are you able to GPTQ the whole thing?
> > > > > 
> > > > > 
> > > > > I tried to load it in CPU it loaded successfully but when I tried to get the output then it is showing the error. I'm using TheBloke/Falcon-180B-Chat-GPTQ this model
> > > > 
> > > > 
> > > > If you load this model in CPU you should install Obaboga in CPU mode only and you should use GGUF model variant for CPU mode.
> > > 
> > > 
> > > GGUF is too large. And how to load obaboga in CPU ?
> > 
> > 
> > Install Obaboga from a scratch and select to install it when isked with CPU mode only.
> 
> Are you a rookie in this? We just need to clone the repository and start the server file. There's no option to load it in CPU.

I'm average in this. You should download [oobabooga_windows.zip](https://github.com/oobabooga/text-generation-webui/releases/download/installers/oobabooga_windows.zip) and unpack files to your folder and click the bat file and it will start installing automatically and during this process it will ask you what mode you want.

**Comment by mabushey** [comment]
This is the error I get on Arch Linux, 128GB RAM, 24GB VRAM:
```
Traceback (most recent call last): File “/app/server.py”, line 68, in load_model_wrapper shared.model, shared.tokenizer = load_model(shared.model_name, loader) File “/app/modules/models.py”, line 78, in load_model output = load_func_maploader File “/app/modules/models.py”, line 148, in huggingface_loader model = LoaderClass.from_pretrained(Path(f"{shared.args.model_dir}/{model_name}"), low_cpu_mem_usage=True, torch_dtype=torch.bfloat16 if shared.args.bf16 else torch.float16, trust_remote_code=shared.args.trust_remote_code) File “/venv/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py”, line 493, in from_pretrained return model_class.from_pretrained( File “/venv/lib/python3.10/site-packages/transformers/modeling_utils.py”, line 2474, in from_pretrained raise EnvironmentError( OSError: Error no file named pytorch_model.bin, tf_model.h5, model.ckpt.index or flax_model.msgpack found in directory models/Falcon-180B-Chat-GPTQ.```

**Comment by mabushey** [comment]
The error I just posted above was a stupid mistake, I missed linking `model.safetensors.index.json` into `config/models/Falcon-180B-Chat-GPTQ`. Now I get this in the logs:
```
2023-09-15 17:47:42 INFO:Loading Falcon-180B-Chat-GPTQ...
/scripts/docker-entrypoint.sh: line 69:    88 Killed                  "${LAUNCHER[@]}"
```

**Comment by mabushey** [comment]
Restarting everything, I now get:
```
Traceback (most recent call last): File “/app/server.py”, line 68, in load_model_wrapper shared.model, shared.tokenizer = load_model(shared.model_name, [loader](http://localhost:7860/model_name)) File “/app/modules/models.py”, line 78, in load_model output = load_func_maploader File “/app/modules/models.py”, line 287, in AutoGPTQ_loader return modules.AutoGPTQ_loader.load_quantized(model_name) File “/app/modules/AutoGPTQ_loader.py”, line 56, in load_quantized model = AutoGPTQForCausalLM.from_quantized(path_to_model, **params) File “/venv/lib/python3.10/site-packages/auto_gptq/modeling/auto.py”, line 85, in from_quantized model_type = check_and_get_model_type( File “/venv/lib/python3.10/site-packages/auto_gptq/modeling/_utils.py”, line 147, in check_and_get_model_type raise TypeError(f"{config.model_type} isn’t supported yet.") TypeError: falcon isn’t supported yet.```

Does anyone have an estimation on when falcon will be supported?

**Comment by sirus20x6** [comment]
I tried all the loaders and none work for the gguf

**Comment by Ph0rk0z** [comment]
Uh.. you can't load a GPTQ model to CPU. Haven't tried that version but GGUF runs for me fine with correctly set up python bindings. Seems there is nothing special about falcon 180b. Also word of warning, even if they fix offloading for AutoGPTQ (someone did commit a CPU kernel) the last I checked their multi-gpu support was not great. 

Best way to run falcon is to use the openAI rest server in llama CPP python bindings.

**Comment by sirus20x6** [comment]
I've tried the gguf version and I can't get it to work with any loader. Am I supposed to unsplit it from the 4 files?

**Comment by Kunjesh07** [comment]
> I've tried the gguf version and I can't get it to work with any loader. Am I supposed to unsplit it from the 4 files?

Try to combine all files in one file.

**Comment by Ph0rk0z** [comment]
Right, it must be concatenated. those .a .b files won't load by themselves.

**Comment by Flanua** [comment]
All the split files for your model should be combined! No wonder for some of you the model didn't even load. People should read the instruction first.

**Comment by sirus20x6** [comment]
I downloaded from the bloke. I looked at every letter of the model card and there were no such instructions. also copy is not a linux program

**Comment by Flanua** [comment]
> I downloaded from the bloke. I looked at every letter of the model card and there were no such instructions. also copy is not a linux program

Look again because It's literally in the middle of the model card and it says:
Click for instructions regarding joining files
